src/core/auth/qgsauthmanager.cpp

```diff
@@ -84,6 +84,10 @@ const QString QgsAuthManager::AUTH_PASSWORD_HELPER_DISPLAY_NAME( "Password Manag
 
 QgsAuthManager *QgsAuthManager::instance()
 {
+  // NOTE cppcheck complains about identicalInnerCondition. Pedantically
```

Comment:
Could we replace this with a std::once?

Reply:
That would certainly be the most simple solution to make that correct compared to the other alternatives with the memory ordering hints. There's a small overhead though when using std::once . Not sure how it would compared to just removing the initial check and going through the mutex systematically

Reply:
Fair point! In this case no performance critical code should be interacting with the auth manager, so it would be safe to use here. But whatever's easiest!

Reply:
changed to systematically going through the mutex